torch/distributed/fsdp/flat_param.py

```diff
@@ -1487,7 +1487,9 @@ def reshard(self, free_unsharded_flat_param: bool):
         """
         Runs the reshard logic. This includes freeing the unsharded flat
         parameter if ``free_unsharded_flat_param`` and switching to using the
-        sharded flat parameter.
+        sharded flat parameter. Note that this also implicitly offloads
+        the sharded flat parameter (if CPU offload is enabled) by pointing
+        it to the local_shard attribute which resides on CPU.
```

Comment:
nit: Should we specify _local_shard with the prefixed underscore? (backticks do not matter much can be can do \'\'_local_shard\'\' for consistency)